static/src/javascripts/projects/common/modules/audio/AudioPlayer.js

```diff
@@ -295,13 +276,16 @@ export class AudioPlayer extends Component<Props, State> {
             duration: 0,
             hasBeenPlayed: false,
             waveWidthPx: 0,
+            hovering: false,
+            grabbing: false,
         };
     }
 
     componentDidMount() {
         this.audio.addEventListener('timeupdate', this.onTimeUpdate);
         // this should fire on Firefox
         this.audio.addEventListener('ended', this.resetAudio);
+        this.audio.addEventListener('progress', this.buffer);
 
         if (Number.isNaN(this.audio.duration)) {
```

Comment:
Just wondering - will this code ever be fired - if we are using the number duration from the scala template? We can probably delete it..

Reply:
Good question. Yes it will - `props.duration` is the length of the audio file without ads, so we have no other choice but to check `audio.duration` when it is available.